airbyte_cdk/sources/declarative/partition_routers/async_job_partition_router.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from dataclasses import InitVar, dataclass, field
+from typing import Any, Callable, Iterable, Mapping, Optional
+
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.declarative.async_job.job_orchestrator import (
+    AsyncJobOrchestrator,
+    AsyncPartition,
+)
+from airbyte_cdk.sources.declarative.partition_routers.single_partition_router import (
+    SinglePartitionRouter,
+)
+from airbyte_cdk.sources.streams.concurrent.partitions.stream_slicer import StreamSlicer
```

Comment:
This was intentional to use the concurrent `StreamSlicer` interface. There is probably an argument for consistency to use the declarative `StreamSlicer`, but that would require we implement unused methods for providing request parameters and we expect the concurrent interface to be the long term interface.